returns/tracing.py

```diff
@@ -0,0 +1,42 @@
+import types
+from contextlib import contextmanager
+from inspect import FrameInfo, stack
+from typing import List, Optional
+
+from returns.result import _Failure
+
+
+@contextmanager
+def collect_traces():
+    """
+    Context Manager/Decorator to active traces collect to the Failures.
+
+    .. code:: python
+
+        >>> from inspect import FrameInfo
+
+        >>> from returns.io import IOResult
+        >>> from returns.result import Result
+        >>> from returns.tracing import collect_traces
+
+        >>> with collect_traces():
+        ...     traced_failure = Result.from_failure('Traced Failure')
+        >>> non_traced_failure = IOResult.from_failure('Non Traced Failure')
+
+        >>> assert non_traced_failure.trace is None
+        >>> assert isinstance(traced_failure.trace, list)
```

Comment:
Can we make this value visual? Let's not use `assert` here and cast it to text format.
This way user would know what to expect.

Reply:
One discussion we didn't have about **trace** property, should it return the list of *FrameInfo* or print out the list properly formated?

Reply:
It should return a `list`, this more like FP 🙂